### ISSUE TITLE
[p-fileUpload] Unable to select another file after either canceling or uploading on IE11 with advanced file uploader

### ISSUE THREAD

**Issue by dkillack** [primefaces/primeng#4008]
**I'm submitting a ...**  (check one with "x")
```
[ x ] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**

http://plnkr.co/edit/vonmEQDF6ZjHxp6xOxPP?p=preview
This also happens on the FileUpload documentation page: https://www.primefaces.org/primeng/#/fileupload

**Current behavior**
On IE11: Attempting to select another file to upload after either uploading or canceling fails using the advanced file uploader.

**Expected behavior**
On IE11: User can re-upload a different file after canceling or uploading a document using the advanced file uploader.

**Minimal reproduction of the problem with instructions**
STEPS TO REPRODUCE (in IE11, using the provided plunkr link):
- Select Browse and choose an image to upload
- Select either Cancel or Upload
- Select Browse and choose a different image
- Result: No file is selected.

**Angular version:** 4.2.6

**PrimeNG version:** 4.2.1

**Browser:** IE11

**Comment by dkillack** [comment]
I believe this has to do with the latest commit: https://github.com/primefaces/primeng/commit/54f277c3de491efd8584eb93acc2019afde73359. When the user selects a file, the selfInputChange flag goes from undefined to true (with first event) then to false (with second event). But then when the user selects Cancel or Upload, clearInputElement gets called again which sets that value to true, thus not allowing the user to upload immediately again (that just turns the flag false). I think the flag management might be tricky here: any reason to not just stop the propagation for the first event, thus preventing the second?

**Comment by ogousa** [comment]
It's not the best solution, but it can be fixed by adding one parameter with default value = false to clearInputElement(), like this:

```
clearInputElement(flag: boolean = false) {
	if (this.advancedFileInput && this.advancedFileInput.nativeElement) {
		if (this.isIE11()) {
			this.selfInputChange = flag; //IE11 fix to prevent onFileChange trigger again
		}

		this.advancedFileInput.nativeElement.value = '';
	}
}

```

and passing true only from onFileSelect(), like this:

`this.clearInputElement(true);`


Tested locally in IE11 - it works.

**Comment by bnymncoskuner** [comment]
We're having the same problem. Is there any plan to fix this in short term or can I do anything to fix this without changing the primeng code itself until the official fix?

**Comment by cagataycivici** [comment]
Unable to replicate, please provide a test case based on plunkr below and we'll review again.

http://plnkr.co/edit/6q5v84DWolH6DeAdi70w?p=info

**Comment by bnymncoskuner** [comment]
I'm able to replicate this on primeng demo page. https://www.primefaces.org/primeng/#/fileupload
My IE version is 11.0.9600.18793
Choose a file and clear it (either with cancel button or X button next to the file)
Afterwards, you cannot choose another file.

On drag, after deletion the first file, you cannot drag another one at first try however if you try one more time it works just fine.

**Comment by HasanTariq** [comment]
Hi, 
I have just come across the same issue and my IE version is 11.0.9600.18816

I though the problem is somewhat related to Angular version but as @bnymncoskuner mentioned when replicated the issue on priming demo page.

**Comment by cholguin86** [comment]
Hi!. I have the same issue. I have IE 11.0.9600.17801. You can reproduce it with the PrimeNG online demo (https://www.primefaces.org/primeng/#/fileupload) as mentioned bnymncoskuner. Steps: Load a file previously loaded or load a file previously removed from list, the file is not shown.

My  used versions:
Angular: 4.2.4
PrimeNG: 4.3.0

Any solution for this issue?. Thank you!

**Comment by cholguin86** [comment]
I could solve my issue implementing the following methods:

`onRemove(event: any) {           
    if (this.fileUpload.isIE11()) {                       
        this.fileUpload.selfInputChange = false;              
    }
  }
  
  onClear(event: any) {
    if (this.fileUpload.isIE11()) {                       
        this.fileUpload.advancedFileInput.nativeElement = null;
        this.fileUpload.selfInputChange = false;              
    }
  }`

I have seen the managment of events is very different on this browser, so I am forcing the cleaning of the component.


At my template I have:

`(onBeforeSend)="onBeforeSend($event)" (onClear)="onClear($event)"`

**Note:** this.fileUpload is a binding of the "p-fileUpload", for my case I declared `@ViewChild(FileUpload) fileUpload: FileUpload;`

**Comment by bnymncoskuner** [comment]
I have tried @cholguin86 solution and it works for now. I'll keep it until the official fix.

Did you have any chance to check this one out? If so, is there any plan to work on this issue or should we open another one? @cagataycivici

**Comment by bandersen22000** [comment]
I have the same error for IE 11 with "primeng": "^5.0.2",

**Comment by dkillack** [comment]
Might have better luck with these tickets as they have not been closed: #4273, #4733.
